etcd/v1alpha1/types_pacemakerstatus.go

```diff
@@ -0,0 +1,312 @@
+	// +kubebuilder:validation:MaxLength=2048
+	// +optional
+	CollectionError string `json:"collectionError,omitempty"`
+
+	// summary provides high-level counts and flags for the cluster state
+	// +optional
+	Summary *PacemakerSummary `json:"summary,omitempty"`
+
+	// nodes provides detailed information about each node in the cluster
+	// +listType=atomic
+	// +kubebuilder:validation:MinItems=2
+	// +kubebuilder:validation:MaxItems=2
+	// +optional
+	Nodes []NodeStatus `json:"nodes,omitempty"`
+
+	// resources provides detailed information about each resource in the cluster
+	// +listType=atomic
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=16
+	// +optional
+	Resources []ResourceStatus `json:"resources,omitempty"`
+
+	// nodeHistory provides recent operation history for troubleshooting
+	// +listType=atomic
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=16
+	// +optional
+	NodeHistory []NodeHistoryEntry `json:"nodeHistory,omitempty"`
+
+	// fencingHistory provides recent fencing events
+	// +listType=atomic
+ 	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=16
+	// +optional
+	FencingHistory []FencingEvent `json:"fencingHistory,omitempty"`
+}
+
+// PacemakerSummary provides a high-level summary of cluster state
+type PacemakerSummary struct {
+	// pacemakerdState indicates if pacemaker is running
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=16
+	// +optional
+	PacemakerdState string `json:"pacemakerdState,omitempty"`
+
+	// quorumStatus indicates if the cluster has quorum
+	// +optional
+	QuorumStatus QuorumStatusType `json:"quorumStatus,omitempty"`
+
+	// nodesOnline is the count of online nodes
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=2
+	// +optional
+	NodesOnline *int32 `json:"nodesOnline,omitempty"`
+
+	// nodesTotal is the total count of configured nodes
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=2
+	// +optional
+	NodesTotal *int32 `json:"nodesTotal,omitempty"`
+
+	// resourcesStarted is the count of started resources
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=16
+	// +optional
+	ResourcesStarted *int32 `json:"resourcesStarted,omitempty"`
+
+	// resourcesTotal is the total count of configured resources
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=16
+	// +optional
+	ResourcesTotal *int32 `json:"resourcesTotal,omitempty"`
+}
+
+// NodeStatus represents the status of a single node in the Pacemaker cluster
+type NodeStatus struct {
+	// name is the name of the node
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=256
+	// +optional
+	Name string `json:"name,omitempty"`
+
+	// ip is the IP address of the node
+	// +kubebuilder:validation:MinLength=2
+	// +kubebuilder:validation:MaxLength=62
+	// +optional
+	IP string `json:"ip,omitempty"`
```

Comment:
Do we want to enforce a tighter validation using a regex here? (via `+kubebuilder:validation:Pattern`)